src/xenia/gpu/dxbc_shader_translator.cc

```diff
@@ -435,12 +435,19 @@ void DxbcShaderTranslator::StartVertexOrDomainShader() {
                                                   : dxbc::Dest::R(0, 0b0111),
                  dxbc::Src::VDomain(0b000110));
         if (register_count() >= 2) {
-          // Copy the primitive index to r1.x as a float.
+          // Copy the primitive index + offset to r1.x as a float.
           uint32_t primitive_id_temp =
               uses_register_dynamic_addressing ? PushSystemTemp() : 1;
           in_primitive_id_used_ = true;
+          a_.OpMov(dxbc::Dest::R(primitive_id_temp, 0b0001),
+                   dxbc::Src::VPrim());
+          a_.OpIAdd(dxbc::Dest::R(primitive_id_temp, 0b0001),
+                    dxbc::Src::R(primitive_id_temp, dxbc::Src::kXXXX),
+                    dxbc::Src::CB(cbuffer_index_system_constants_,
+                                  uint32_t(CbufferRegister::kSystemConstants),
+                                  1, dxbc::Src::kYYYY));
           a_.OpUToF(dxbc::Dest::R(primitive_id_temp, 0b0001),
```

Comment:
As there are multiple instances of this code, and with clamping it will be even longer, it can be moved to `void LoadTessellationPatchIndex(uint32_t dest_temp, uint32_t dest_temp_component)` (see other functions accepting temporary register index/component in `DxbcShaderTranslator` for reference, such as 20e4/7e3 float conversion).

Or better, even to a function that would also be used for the vertex index after byte swapping (accepting the source as the third `const dxbc::Src& src` argument, which can safely be the same as the destination as it will be referenced only once, in the first operation), like `ConvertVertexIndex`, doing offsetting, clamping, and float32 conversion.